### PR TITLE
Fix lint issues when bumping eslint-plugin-import

### LINKED ISSUE #1613: [deps] Help with upgrading eslint-plugin-import
@dependabot tried to upgrade https://github.com/jaegertracing/jaeger-ui/pull/1609 but it broke the CI, probably requires some minor code changes to satisfy the new linter version

### PR DESCRIPTION
**Pull request by Kavinjsir**
## Which problem is this PR solving?
- Resolves #1613
- Part of #1608

## Description of the changes
- Update code lines to resolve lint check warnings

## How was this change tested?
- `yarn run esilnt`

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality

### CODE REVIEW

File: packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 Uber Technologies, Inc.
+// Copyright (c) 2023 Uber Technologies, Inc.
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js:1]: why is this changed?

File: packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.tsx [hunk 2]
```diff
@@ -93,6 +93,11 @@ const SelectSort = reduxForm({
 
 export const sortFormSelector = formValueSelector('traceResultsSort');
 
+// export for tests
+export function createBlob(rawTraces: TraceData[]) {
+  return new Blob([`{"data":${JSON.stringify(rawTraces)}}`], { type: 'application/json' });
+}
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.tsx:99]: this looks just like a move, what is the purpose?

File: packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js (hunk 1 above)
Comment (Kavinjsir) [reply packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js:1]: I am wondering if we need to keep the year up to date if we have the code lines in the file get updated?
Or do we keep them as origin?

File: packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.tsx (hunk 2 above)
Comment (Kavinjsir) [reply packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.tsx:99]: So the update of the code lines is an attempt to resolve the lint warnings.
For this one, before the update, running `yarn run eslint` will result in:
```
/Users/jintony/pro/tmp/jaeger-ui/packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.tsx
  117:18  warning  'createBlob' was used before it was defined  @typescript-eslint/no-use-before-define
```

That's why I lift the definition up.

File: packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js (hunk 1 above)
Comment (yurishkuro) [reply packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js:1]: Uber's was the original copyright, new contributions should use "The Jaeger Authors", which you can add as a line above with the current year if you want.

We tend to keep the original year a specific (c) by-line was introduced. I checked with legal once, there wasn't a lot of value in updating the year.

**Comment by Kavinjsir** [comment]
I think I might miss the point of ci error before. Just realized that `unit-test` in ci failed by running `yarn lint`:
https://github.com/jaegertracing/jaeger-ui/actions/runs/5815079038/job/15766674298

It seems we can reproduce the issue locally by removing the *--cache* flag when running *eslint*. Let me try resolving them.

File: packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js (hunk 1 above)
Comment (Kavinjsir) [reply packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js:1]: Sounds good! I'm going to roll back the corresponding change as it is not necessary. Thx for letting me know:)

File: packages/jaeger-ui/src/demo/trace-generators.js [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 import Chance from 'chance';
 
-import { getSpanId } from '../selectors/span';
+import getSpanId from '../selectors/span';
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/demo/trace-generators.js:17]: this is an exceedingly silly selector, is it actually used, or can we delete it?

File: packages/jaeger-ui/src/selectors/trace.test.js [hunk 4]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 import { followsFromRef } from './trace.fixture';
-import { getSpanId } from './span';
+import getSpanId from './span';
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/selectors/trace.test.js:16]: if this is the only place, let's just replace it with span.ID and delete the selector

**Comment by yurishkuro** [comment]
your description links to issues, but does not put the word Fixes or Resolves in front of them. Please update, this way when the PR is merged the issues will be closed (unless it does not resolve them completely, in which case we typically use the format `Part of #xxxx`

File: package.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
     "eslint": "8.46.0",
     "eslint-config-airbnb": "19.0.4",
     "eslint-config-prettier": "9.0.0",
-    "eslint-plugin-import": "2.27.5",
+    "eslint-plugin-import": "2.28.0",
```
Comment (yurishkuro) [inline package.json:17]: once the CI is green, I would prefer to remove the version upgrade and only leave the code changes, then let the dependabot to redo its PR for the actual upgrade.

**Review by yurishkuro — DISMISSED** [review]

**Comment by yurishkuro** [comment]
@Kavinjsir I removed the upgrade to keep the scope to code changes only, and it fails the CI, can you please check why?

File: packages/jaeger-ui/src/selectors/trace.test.js (hunk 4 above)
Comment (Kavinjsir) [reply packages/jaeger-ui/src/selectors/trace.test.js]: Sounds good!

**Review by yurishkuro — APPROVED** [review]

File: packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js [hunk 6]
```diff
@@ -175,8 +175,11 @@ describe('<SearchResults>', () => {
 
       it('when click on DownloadResults then call download function', () => {
         const originalBlob = global.Blob;
-        global.Blob = function (text, options) {
-          return { text, options };
+        global.Blob = class {
+          constructor(text, options) {
+            this.text = text;
+            this.options = options;
+          }
```
Comment (Kavinjsir) [inline packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js:182]: @yurishkuro From my local, it seems like what really brought the `unit-test` failure is:
Screenshot 2023-08-10 at 12 47 34 PM

(Not sure why it was missed in the CI logs)

For this error above, I think I made a mistake to convert `Blob` to an arrow function that is not able to call by `new`. 
 
So I change such `global.Blob` to an "explicit" class. Assuming this can:
1. Still resolve the origin lint warning
2. Attempt to make it clear so satisfy calling by keyword `new`.

How do you think of this?

**Comment by yurishkuro** [comment]
Thanks, @Kavinjsir, the dependabot PR #1609 was successfully merged after these changes.
